src/python/gem5/components/boards/kernel_disk_workload.py

```diff
@@ -182,7 +201,8 @@ def set_kernel_disk_workload(
         self.workload.command_line = (
             " ".join(kernel_args or self.get_default_kernel_args())
         ).format(
-            root_value=self.get_default_kernel_root_val(disk_image=disk_image)
+            root_value=self.get_default_kernel_root_val(disk_image=disk_image),
+            disk_device=self.get_disk_device(),
```

Comment:
Ok, we're making progress on this! How about we improve this even more. Why don't we remove `get_disk_device` function entirely and move the logic down into `set_kernel_disk_workload`.